packages/react/src/theme/theme.tsx

```diff
@@ -1,50 +1,53 @@
-import './styles/theme-dx-dark.scss';
-import './styles/variables-dark.scss';
-import './styles/variables-light.scss';
-import './styles/theme-dx-light.scss';
 import { currentTheme as currentVizTheme, refreshTheme } from 'devextreme/viz/themes';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
 
 const themeNames = ['dark', 'light'] as const;
-const themeStylesSheets = {};
-const storageKey = 'themeViewer';
+const storageKey = 'app-theme';
+const themePrefix = 'app-theme-';
 
-export type Theme = typeof themeNames[number];
+const prefixes = ['./styles/theme-dx-', './styles/variables-'];
 
-function getCurrentTheme(): Theme {
-  return window.localStorage[storageKey] || 'light';
-}
+const loadStylesImports = async() => {
+  await Promise.all([
+    ...prefixes.flatMap((prefix) => [
+      import(/* webpackChunkName: "app-theme-dark" */ `${prefix}dark.scss`),
+      import(/* webpackChunkName: "app-theme-light" */ `${prefix}light.scss`)
+    ]),
+  ]);
+};
+
+export type Theme = typeof themeNames[number];
 
 function toArray<T>(arrayLikeObject) {
   return [...arrayLikeObject as unknown as T[]];
 }
 
-function getThemeStyleSheets(): CSSStyleSheet[] {
-  const themePrefix = 'app-theme-';
-
-  if (Object.keys(themeStylesSheets).length == themeNames.length) {
-    return Object.values<CSSStyleSheet>(themeStylesSheets).flat();
-  }
+function getCurrentTheme(): Theme {
+  return window.localStorage[storageKey] || 'light';
+}
 
-  themeNames.forEach((themeName) => {
-    if (themeStylesSheets[themeName]) {
-      return;
-    }
+function isThemeStyleSheet(styleSheet, theme: Theme) {
+  const themeMarker = `${themePrefix}${theme}`;
+  // eslint-disable-next-line no-undef
+  return process.env.NODE_ENV === 'production' ?
+    styleSheet?.href?.includes(`${themeMarker}`)
+    : -1 !== [0, -1].findIndex(
+      (i) => toArray<CSSStyleRule>(styleSheet.cssRules).at(i)?.selectorText?.includes(`.${themeMarker}`)
+    );
+}
 
-    const themeCssSelector = `.${themePrefix + themeName}`;
+function switchThemeStyleSheets(enabledTheme: Theme) {
+  const disabledTheme = enabledTheme === 'dark' ? 'light' : 'dark';
 
-    themeStylesSheets[themeName] = toArray<CSSStyleSheet>(document.styleSheets).filter(
-      ({ cssRules })=> !!toArray<CSSStyleRule>(cssRules).find(({ selectorText }) => selectorText?.includes(themeCssSelector)));
+  return toArray<CSSStyleSheet>(document.styleSheets).forEach((styleSheet) => {
```

Comment:
Do we need to return a value here? It seems it is not used anywhere.

Reply:
fix